app/Http/Controllers/AdminController.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Interfaces\ReportServiceInterface;
+use App\Models\Category;
+use App\Models\Order;
+use App\Models\Product;
+use App\Models\ProductImage;
+use ArielMejiaDev\LarapexCharts\LarapexChart;
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Storage;
+
+class AdminController extends Controller
+{
+    protected $reportService;
+
+    public function __construct(ReportServiceInterface $reportService)
+    {
+        $this->reportService = $reportService;
+
+        
+
+    }
+
+    public function index()
+    {
+        $reportData = $this->reportService->index();
+        return view('admin.index', $reportData);
+    }
+
+    public function productIndex()
+    {
+        $products = Product::all();
+        return view('admin.products.index', compact('products'));
+    }
+
+    public function createProduct()
+    {
+        $categories = Category::all();
+        return view('admin.products.create', compact('categories'));
+    }
+
+    public function storeProduct(Request $request)
+    {
+        $validated = $request->validate([
+            'title' => 'required|string|max:100',
+            'description' => 'required|string',
+            'price' => 'required|numeric',
+            'category_id' => 'required|exists:categories,id',
+            'discount' => 'nullable|integer',
+            'images.*' => 'image|mimes:jpeg,png,jpg,gif|max:2048',
+            'default_photo' => 'nullable|exists:product_images,id',
+        ]);
+
+        $product = Product::create([
+            'title' => $validated['title'],
+            'description' => $validated['description'],
+            'price' => $validated['price'] * 100,
+            'category_id' => $validated['category_id'],
+            'discount' => $validated['discount'],
+            'photo' => '',
+        ]);
+
+        if ($request->hasFile('images')) {
+        $imagePaths = [];
+        $first = true;
+
+        foreach ($request->file('images') as $index => $image) {
+            $path = $image->store('product-photos', 'public');
+            $imagePaths[$index] = $path;
+
+            $isDefault = ($request->input('default_photo') == $index) || ($first && !$request->has('default_photo'));
+
+            ProductImage::create([
+                'product_id' => $product->id,
+                'image_path' => $path,
+                'is_default' => $isDefault,
+            ]);
+
+            if ($isDefault) {
+                $product->update(['photo' => $path]);
```

Comment:
ფოტოს path ის შენახვას, აიდი შეინახე ფოტოსი რო ორ ადგილას არ იყოს "სიმართლის წყარო" (source of truth)